tensorflow_datasets/image/breastpathq.py

```diff
@@ -0,0 +1,132 @@
+"""Breast cancer whole slide image dataset."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import os
+import csv
+import numpy as np
+
+import tensorflow as tf
+import tensorflow_datasets as tfds
+
+_URL = "http://spiechallenges.cloudapp.net/competitions/14#participate"
+
+# BibTeX citation
+_CITATION = """\
+@article{peikari2017automatic,
+  title={Automatic cellularity assessment from \
+  post-treated breast surgical specimens},
+  author={Peikari, Mohammad and Salama, Sherine and \
+  Nofech-Mozes, Sharon and Martel, Anne L},
+  journal={Cytometry Part A},
+  volume={91},
+  number={11},
+  pages={1078--1087},
+  year={2017},
+  publisher={Wiley Online Library}
+}
+"""
+
+
+_DESCRIPTION = """\
+The dataset's training/validation set consists of \
+2578 patches extracted from 96 breast cancer \
+whole slide images (WSI). Each patch is labelled \
+by a tumor cellularity score. The testing set \
+contains 1121 patches from 25 WSIs. Labels for \
+testing data are not provided by far. \
+The dataset can be used to develop an automated method \
+for evaluating cancer cellularity from \
+histology patches extracted from WSIs. The method \
+is aimed to increase reproducibility of cancer \
+cellularity scores and enhance tumor burden assessment.
+"""
+
+_IMAGE_SHAPE = (512, 512, 3)
+
+def _load_tif(path):
+  with tf.io.gfile.GFile(path, "rb") as fp:
+    image = tfds.core.lazy_imports.PIL_Image.open(fp)
+    rgb_img = image.convert("RGB")
+  return np.array(rgb_img)
+
+
+class Breastpathq(tfds.core.GeneratorBasedBuilder):
+  """Breast cancer whole slide image dataset."""
+
+  # Set up version.
+  VERSION = tfds.core.Version('0.1.0')
+
+  def _info(self):
+    # Specifies the tfds.core.DatasetInfo object
+    return tfds.core.DatasetInfo(
+        builder=self,
+        # This is the description that will appear on the datasets page.
+        description=_DESCRIPTION,
+        # tfds.features.FeatureConnectors
+        features=tfds.features.FeaturesDict({
+            # These are the features of your dataset like images, labels ...
+            "image": tfds.features.Image(),
+            "label": tf.float32
+        }),
+        # If there's a common (input, target) tuple from the features,
+        # specify them here. They'll be used if as_supervised=True in
+        # builder.as_dataset.
+        supervised_keys=("image", "label"),
+        # Homepage of the dataset for documentation
+        urls=[_URL],
+        citation=_CITATION,
+    )
+
+  def _split_generators(self, dl_manager):
+    """Returns SplitGenerators."""
+    # Downloads the data and defines the splits
+    # dl_manager is a tfds.download.DownloadManager that can be used to
+    # download and extract URLs
+    # manual download is required for this dataset
+    download_path = dl_manager.manual_dir
+    extracted_path = dl_manager.extract(download_path)
```

Comment:
Since the download path is a directory, the extraction doesn't occur. This should be path to files otherwise the `extracted_path` is just the same as `download_path`.